### PR TITLE
chore(#3774): add integration tests for connection query

### LINKED ISSUE #3774: Cleanup ConnectionQuery | Add integration tests for ConnectionQuery correctness
## Problem

There are no integration tests that verify ConnectionQuery results across consumer scenarios. Refactoring without a safety net risks silent regressions.

## Goal

A test suite that asserts ConnectionQuery behavior for representative filter combinations, serving as a regression safety net for all subsequent refactoring.

## Work

- Integration tests covering:
  - AuthorizedParties: `FromOthers` with packages, exclude deleted, entity enrichment
  - PIP: `FromOthers` with packages + package resources, no entity enrichment
  - Connection API: both directions, full enrichment (entities, packages, resources, instances)
  - `HasConnection`: verify true/false for each connection reason (Assignment, Delegation, Hierarchy, KeyRole)
  - Edge cases: subunit/mainunit hierarchy, ENK/Innehaver, IKS exclusion, large FromId sets
- Tests run against the current (unmodified) implementation

## Out of scope

- Refactoring any production code
- Performance benchmarks

## Constraints

- Tests must pass on `main` before any refactoring begins
- Use existing test infrastructure and database seeding patterns

### PR DESCRIPTION
**Pull request by jonkjetiloye**
### Summary

Adds a comprehensive integration test suite for `ConnectionQuery` as a regression safety net before refactoring begins (#3756). All tests run against the current unmodified implementation and must pass on `main`.

### What's included

**New seed data** added to `TestDataSet`:
- ENK (Enkeltpersonforetak) entity with Innehaver relationship
- Consumer & Supplier Corp entities with Supplier-role assignment
- AppControlledRightholder assignment
- ResourceType, Resource, AssignmentResource, and AssignmentInstance records
- AssignmentPackages and DelegationPackages (were defined but not seeded before)

**new test methods** covering:

| Category | Tests | What's verified |
|---|---|---|
| **HasConnection – Assignment** | 2 | Direct assignment returns `Assignment` reason; no assignment returns false |
| **HasConnection – Delegation** | 2 | Delegation returns `Delegation` reason; no delegation returns false |
| **HasConnection – Hierarchy** | 2 | Subunit via parent returns `Hierarchy`; unrelated subunit returns false |
| **HasConnection – Priority** | 2 | Multiple matching reasons returns highest priority; empty reasons returns false |
| **AuthorizedParties (packages)** | 2 | Delegation packages (3 accountant packages for Gunnar); assignment packages (AOrderSystem for Nina) | | **PIP scenario** | 1 | Packages returned without entity enrichment |
| **ToOthers direction** | 1 | Regnskaperne → returns Petter and Gunnar |
| **ToOthers from subunit** | 1 | Baker Johnsen - Oslo inherits mainunit connections via Hierarchy |
| **ToOthers + delegations** | 2 | Client delegations appear in ToOthers; absent when `IncludeDelegation=false` |
| **FromOthers & ToOthers + Supplier assignment** | 2 | Supplier-assignmenst should not appear in either ToOthers or FromOthers |
| **ENK/Innehaver** | 1 | Innehaver appears as connection via ENK for revi/regn relationships |
| **RoleMap connections** | 1 | ManagingDirector produces `RoleMap` results from static-data role maps |
| **IncludeResources** | 2 | Resources loaded on Rightholder assignments; not loaded when flag is off |
| **IncludeInstances** | 1 | Instances loaded on Rightholder assignments |
| **PackageIds filtering** | 2 | Only matching packages returned; connections with no match removed | | **Supplier exclusion** | 1 | Supplier role never appears in results |
| **IncludeSubConnections** | 1 | No hierarchy children when flag is false |
| **IncludeMainUnitConnections** | 1 | No mainunit assignments when flag is false |
| **AppControlledRightholder** | 2 | Excluded by default; included when `IncludeAppControlledInstances=true` |

### No production code changes

All changes are in the test project only.

### Related issues

- Part of #3756 (Cleanup ConnectionQuery)
- Closes #3774 (Add integration tests for ConnectionQuery correctness)

### CODE REVIEW

File: src/apps/Altinn.AccessManagement/test/AccessMgmt.Tests/Integration/Services/ConnectionQueryTests.cs [hunk 1]
```diff
@@ -446,6 +456,667 @@ public async Task KeyRole_ToOthers_IksWithDifferentRole_IsIncluded()
...
+        {
+            ToIds = new[] { personId },
+            IncludeKeyRole = true,
+            IncludeDelegation = true,
+            IncludeSubConnections = false,
+            EnrichEntities = true,
+        };
+
+        var dbResult = await _query.GetConnectionsFromOthersAsync(filter, TestContext.Current.CancellationToken);
+        var connections = DtoMapper.ConvertFromOthers(dbResult, false);
```
[the remark this answers is not in the document]
Comment (jonkjetiloye) [reply src/apps/Altinn.AccessManagement/test/AccessMgmt.Tests/Integration/Services/ConnectionQueryTests.cs]: Fixed

File: src/apps/Altinn.AccessManagement/test/AccessMgmt.Tests/Integration/Services/ConnectionQueryTests.cs [hunk 2]
```diff
@@ -446,6 +456,667 @@ public async Task KeyRole_ToOthers_IksWithDifferentRole_IsIncluded()
...
+        Assert.Contains(dbResult, r => r.ToId == TestDataSet.GetEntity("Petter").Id);
+        Assert.Contains(dbResult, r => r.ToId == TestDataSet.GetEntity("Gunnar").Id);
+    }
+
+    #endregion
+
+    #region ENK/Innehaver
+
+    [Fact]
+    public async Task GetConnectionsFromOthers_EnkInhehaver_ReturnsInnehaverAsConnection()
```
[the remark this answers is not in the document]
Comment (jonkjetiloye) [reply src/apps/Altinn.AccessManagement/test/AccessMgmt.Tests/Integration/Services/ConnectionQueryTests.cs]: Fixed

File: src/apps/Altinn.AccessManagement/test/AccessMgmt.Tests/Integration/Services/ConnectionQueryTests.cs [hunk 3]
```diff
@@ -446,6 +456,667 @@ public async Task KeyRole_ToOthers_IksWithDifferentRole_IsIncluded()
...
+        // Hierarchy check: from a in Assignments where a.ToId == toId, join e where e.ParentId == a.FromId and e.Id == fromId
+        // We need: assignment where ToId=Petter and FromId=Baker Johnsen, then entity Baker Johnsen - Oslo with ParentId=Baker Johnsen
+        // Assignment: Regnskaperne -> Petter (not Baker Johnsen -> Petter)
+        // Actually the hierarchy query is: assignments where a.ToId == toId, joined with entities where e.ParentId == a.FromId and e.Id == fromId
+        // So we need an assignment with ToId=Petter where FromId is the parent of fromId (Baker Johnsen - Oslo's parent = Baker Johnsen)
+        // Assignment: Regnskaperne -> Petter has FromId = Regnskaperne, not Baker Johnsen
+        // Let's use Regnskaperne - Oslo -> Petter path: Regnskaperne has assignment to Petter, Oslo has ParentId = Regnskaperne
+        var fromIdOslo = TestDataSet.GetEntity("Regnskaperne - Oslo").Id;
+
+        var (result, reason) = await _query.HasConnection(fromIdOslo, toId, [ConnectionReason.Hierarchy]);
```
[the remark this answers is not in the document]
Comment (jonkjetiloye) [reply src/apps/Altinn.AccessManagement/test/AccessMgmt.Tests/Integration/Services/ConnectionQueryTests.cs]: removed duplicate fromId

**Review by howieandersen — APPROVED** [review]
LGTM
